### PR TITLE
Don't break page if GCSE doesn't have a grade

### PR DESCRIPTION
**Pull request by carlosmartinez**
## Context

Currently it's not possible for a provider or support user to view an application with a GCSE that doesn't have a grade or constituent_grades; they see a 500 error.

## Changes proposed in this pull request

Instead of erroring, fail gracefully-ish with a warning message.

## Guidance to review

- Update an application to have a science GCSE with `grade: nil, constituent_grades: nil`
- Attempt to access that application in the support console
- Let the good times roll

## Link to Trello card

https://trello.com/c/NBDcyF3b/1204-it-shouldnt-be-possible-for-candidates-to-enter-a-gcse-science-without-grade-or-constituent-grades

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Review by JR-G — APPROVED** [review]

File: app/components/shared/gcse_qualification_cards_component.html.erb [hunk 1]
```diff
@@ -74,7 +74,13 @@
...
                 <% end %>
               <% else %>
-                <% raise 'Neither `grade` nor `constituent_grades` found' %>
+                <div class="govuk-warning-text">
+                  <span class="govuk-warning-text__icon" aria-hidden="true">!</span>
+                  <strong class="govuk-warning-text__text">
+                    <span class="govuk-warning-text__assistive"></span>
+                    There is no grade recorded for this qualification.
+                  </strong>
+                </div>
```
Comment (frankieroberto) [inline app/components/shared/gcse_qualification_cards_component.html.erb:83]: I don't think we need to use the warning text here. A simple message styled the same as the rest, and using a `<dd>` element, should be fine (and more accessible)?

```suggestion
                    <dd class="app-qualification__value">Not added</dd>
```

**Review by frankieroberto — CHANGES_REQUESTED** [review]
Do we know how/why this has happened?

**Comment by carlosmartinez** [comment]
> Do we know how/why this has happened?

In the Bad Old Days, it seems we were missing a validation.

**Review by frankieroberto — APPROVED** [review]

**Comment by frankieroberto** [comment]
> > Do we know how/why this has happened?
> 
> In the Bad Old Days, it seems we were missing a validation.

😞  Does it affect any current (awaiting decision) applications? 🤞

**Pull request merged**